### DISCUSSION TITLE
Levelpack page

### DISCUSSION THREAD

**Discussion by sunehs** [elmadev/elmaonline-site#418] [Feature discussion]
For general discussion of the levelpack page: https://elma.online/levels/packs/Int and https://elma.online/levels

This post will be updated to contain planned features and changes as well as ideas that needs to be fledged out. Discuss them below. If you want to work on one of these, create an issue if it doesn't exist and assign yourself.

**Planned**

**Ideas**
- Improve load time: Some caching to load initially and then reload in the background after getting newest data #422 
- Show personal TT on levelpack list: Use data from caching mentioned above otherwise it's too much data
- Compare: Compare times to kuski, records, wr table X, target times
- Combined kinglists: Total amount of points in all kinglist, weighted points based on rating or popularity, kinglist for collections
- National and team records
- Historic tables, seasonal tables
- stats like old moposite wr stats

**Comment by j-masci** [comment]
Tags and possibly rating types for level pack maintainer to fill out. A rating type could be difficulty or level duration on a scale of 1 to 5. People could then search by these criteria. 

Also a star rating like on replays.

**Comment by j-masci** [comment]
Using the new Level Stats data, we can add to the **archive** levelpack page (https://elma.online/levels)

- Show the top record holders for every pack and the number of records they hold for levels in a pack.
- Show aggregate statistics for all levels in the pack (total time driven, attempts, win/death %, # of players (player count is hard or maybe not doable actually)).
- Have some sorting options to sort by levels most played or possibly other statistics.

**Comment by j-masci** [comment]
On the individual levelpack page (https://elma.online/levels/packs/DCup03). (using new level stats data unless it doesn't require it)
- Show some additional level stats in the table. Something about popularity / time played / kuski's beaten / leaders count.
- Show the top 3 finishers if space permits ? (I think you can do this without new level stats data)
- Add stats tab that aggregates level stats data and shows a lot more than is normally shown on the archive page.

**Comment by j-masci** [comment]
On https://elma.online/levels
- Add a preview levelpack button to each pack. When clicked, a popup opens showing SVG's of first 10 or so levels, with pagination. Since some packs are super huge, we might have to paginate on back-end and only send the visible packs. I'm not sure how much data would get sent for 500-ish levels.
- We could re-use this on the single levelpack page as well.
